spec/predicates/vsa.md

```diff
@@ -1,11 +1,279 @@
+-   The `predicateType` URI includes the major version number and will always
+    change whenever there is a backwards incompatible change.
+-   Minor version changes are always backwards compatible and "monotonic." Such
+    changes do not update the `predicateType`.
+-   Producers MAY add extension fields using field names that are URIs.
+
+### Fields
+
+_NOTE: This section describes the fields within `predicate`. For a description
+of the other top-level fields, such as `subject`, see [Statement]._
+
+<a id="verifier"></a>
+`verifier` _object, required_
+
+> Identifies the entity that performed the verification.
+>
+> The identity MUST reflect the trust base that consumers care about. How
+> detailed to be is a judgment call.
+>
+> Consumers MUST accept only specific (signer, verifier) pairs. For example,
+> "GitHub" can sign provenance for the "GitHub Actions" verifier, and "Google"
+> can sign provenance for the "Google Cloud Deploy" verifier, but "GitHub" cannot
+> sign for the "Google Cloud Deploy" verifier.
+>
+> The field is required, even if it is implicit from the signer, to aid readability and
+> debugging. It is an object to allow additional fields in the future, in case one
+> URI is not sufficient.
+
+<a id="verifier.id"></a>
+`verifier.id` _string ([TypeURI]), required_
+
+> URI indicating the verifier’s identity.
+
+<a id="verifier.version"></a>
+`verifier.version` _map (string->string), optional_
+
+> Map of names of components of the verification platform to their version.
+
+<a id="timeVerified"></a>
+`timeVerified` _string ([Timestamp]), optional_
+
+> Timestamp indicating what time the verification occurred.
+
+<a id="resourceUri"></a>
+`resourceUri` _string ([ResourceURI]), required_
+
+> URI that identifies the resource associated with the artifact being verified.
+
+<a id="policy"></a>
+`policy` _object ([ResourceDescriptor]), required_
+
+> Describes the policy that the `subject` was verified against.
+>
+> The entry MUST contain a `uri`.
+
+<a id="inputAttestations"></a>
+`inputAttestations` _array ([ResourceDescriptor]), optional_
+
+> The collection of attestations that were used to perform verification.
+> Conceptually similar to the `resolvedDependencies` field in [SLSA Provenance].
+>
+> This field MAY be absent if the verifier does not support this feature.
+> If non-empty, this field MUST contain information on _all_ the attestations
+> used to perform verification.
+>
+> Each entry MUST contain a `digest` of the attestation and SHOULD contains a
+> `uri` that can be used to fetch the attestation.
+
+<a id="verificationResult"></a>
+`verificationResult` _string, required_
+
+> Either “PASSED” or “FAILED” to indicate if the artifact passed or failed the policy verification.
+
+<a id="verifiedProperties"></a>
+`verifiedProperties` _array (_string), required_
+
+> Indicates the properties verified for the artifact (and not
+> its dependencies), or "FAILED" if policy verification failed.
```

Comment:
I seem to remember @AdamZWu had some difficulties when implementing this.  Adam is it fine to keep this or would it be better to say the array should be empty?

Reply:
Maybe it's easier to leave it as-is for now so that semantic changes are kept to a minimum?

Reply:
My main reason against having the "FAILED" property is described here: https://github.com/slsa-framework/slsa/issues/968#issuecomment-1733818744

But I am fine for leaving it as a future improvement rather than cram everything is a big change.

> Is this based entirely on the producer<->consumer to reconcile?

My understanding is, sort of. In the context of in-toto specs, yes, in-toto simply provides this field for the producers and consumers, there is no motivation to further regulate its uses. However, under the bigger picture of building a network of secure software supply chains, these are needs for standardization -- If organization A produces "apple" in `verifiedProperties` and organization B expects to see "orange", they cannot effectively exchange the trusted build data.

This, I think, is where SLSA comes in. SLSA defines a set of standardized trusted build properties and levels, which everyone agrees upon *and* mandates that in the VSA, so that, in addition to "apple", organization A must put "SLSA_..." properties in `verifiedProperties`; this makes it possible for organization B to consume that VSA -- they don't know what "apple" means, but they do know what "SLSA_..." means. :D

Reply:
Suggestion: if no one objects lets remove 'FAILED'.  If anyone does object we keep it?